firefox-ios/Client/Frontend/Browser/Toolbars/Redux/ToolbarMiddleware.swift

```diff
@@ -61,6 +61,11 @@ class ToolbarMiddleware: FeatureFlaggable {
         case ToolbarMiddlewareActionType.didTapButton:
             resolveToolbarMiddlewareButtonTapActions(action: action, state: state)
 
+        case ToolbarActionType.showMenuWarningBadge:
+            let action = ToolbarAction(badgeImageName: StandardImageIdentifiers.Large.warningFill,
```

Comment:
follow up to winnie's comment, this seems to make more sense if it live in the state and so the actions don't need to know about this level of detail.